test/contrib/test_enum_elbo.py

```diff
@@ -2472,3 +2472,44 @@ def actual_loss_fn(params_raw):
 
         assert_equal(actual_loss, expected_loss, prec=1e-3)
         assert_equal(actual_grads, expected_grads, prec=1e-5)
+
+
+def test_guide_plate_contraction():
+    def model(params):
+        with pyro.plate("a_axis", size=2):
+            a = pyro.sample("a", dist.Categorical(jnp.array([0.2, 0.8])))
+        pyro.sample("b", dist.Normal(jnp.sum(a), 1.0), obs=1)
```

Comment:
It seems to me that this invalidates this restriction: https://pyro.ai/examples/enumeration.html#Restriction-2:-no-downstream-coupling? Could you elaborate why we can enumerate here?

Reply:
It's not being enumerated here but being used as a non-reparameterizable site. I should have used other distribution like Poisson to make it less confusing :)

But the point is that `b` depends on a non-reparametrizable site `a` which has `a_axis` plate. The `dice_factor` needs to be product-contracted to eliminate the extra `a_axis` plate before multiplying the cost term for site `b`. Instead `a_axis` is being passed to `_eager_contract_tensors` as `reduced_vars` and sum-contracted with `logsumexp`.

Reply:
I see. So we are using TraceEnum_ELBO but enumeration is disabled for those cases. Could you add a warning for this?

Reply:
Do you mean in general or for this test?

Reply:
I mean in general. I think using TraceEnum_ELBO without enumeration is confusing. Maybe raise error if we can't enumerate sites with `infer={"enumerate": "parallel"}`?

Reply:
I changed the distribution to Poisson in the test since it can be any non-reparameterizable distributions. I can open another issue/PR for enumeration configuration since it is a separate issue.

Reply:
sounds reasonable to me, thanks!